source/common/router/router.cc

```diff
@@ -32,6 +32,58 @@ namespace Envoy {
 namespace Router {
 namespace {
 uint32_t getLength(const Buffer::Instance* instance) { return instance ? instance->length() : 0; }
+
+bool schemeIsHttp(const Http::HeaderMap& downstream_headers,
+                  const Network::Connection* connection) {
```

Comment:
I think connection must be non-null through this call chain? If so can you convert to a reference? Same elsewhere where applicable.